trl/trainer/sft_trainer.py

```diff
@@ -240,6 +243,7 @@ def make_inputs_require_grad(module, input, output):
             if data_collator is None:
                 data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=False)
 
+        dataset_kwargs = dataset_kwargs or {}
```

Comment:
```suggestion
        if dataset_kwargs is None:
            dataset_kwargs = {}
```

Reply:
`dataset_kwargs = dataset_kwargs or {}`  should handle this.

Reply:
Yes but I would say let's make it slightly clearer !